nengo/builder/learning_rules.py

```diff
@@ -96,29 +152,36 @@ def get_post_ens(conn):
 @Builder.register(LearningRule)
 def build_learning_rule(model, rule):
     conn = rule.connection
-    rule_type = rule.learning_rule_type
-    pre = get_pre_ens(conn)
-    post = get_post_ens(conn)
-
-    # --- Set up delta signal and += transform / decoders
-    if conn.solver.weights or (
-            isinstance(conn.pre_obj, Neurons) and
-            isinstance(conn.post_obj, Neurons)):
-        delta = Signal(np.zeros((post.n_neurons, pre.n_neurons)), name='Delta')
-        tag = "omega += delta"
-    elif isinstance(conn.pre_obj, Neurons):
-        delta = Signal(np.zeros((rule.size_in, pre.n_neurons)), name='Delta')
-        tag = "omega += delta"
+
+    # --- Set up delta signal
+    if rule.modifies == 'encoders':
+        if not conn.is_factored:
+            ValueError(
+                "Cannot perform encoder learning on an unfactored connection")
```

Comment:
By unfactored do you mean a neuron-to-neuron connection? I think most users will not know what you mean by unfactored.